test/Microsoft.ML.Tests/TrainerEstimators/MatrixFactorizationTests.cs

```diff
@@ -68,20 +68,31 @@ public void MatrixFactorizationSimpleTrainAndPredict()
             var data = reader.Read(new MultiFileSource(GetDataPath(TestDatasets.trivialMatrixFactorization.trainFilename)));
 
             // Create a pipeline with a single operator.
-            var options = new MatrixFactorizationTrainer.Options {
+            var options = new MatrixFactorizationTrainer.Options
+            {
                 MatrixColumnIndexColumnName = userColumnName,
                 MatrixRowIndexColumnName = itemColumnName,
                 LabelColumnName = labelColumnName,
                 NumIterations = 3,
                 NumThreads = 1, // To eliminate randomness, # of threads must be 1.
-                K = 7,
+                ApproximationRank = 7,
             };
 
             var pipeline = mlContext.Recommendation().Trainers.MatrixFactorization(options);
 
             // Train a matrix factorization model.
             var model = pipeline.Fit(data);
 
+            // Let's validate content of the model.
+            Assert.Equal(model.Model.ApproximationRank, options.ApproximationRank);
+            var leftMatrix = model.Model.LeftFactorMatrix;
+            var rightMatrix = model.Model.RightFactorMatrix;
+            Assert.Equal(leftMatrix.Count, model.Model.NumberOfRows * model.Model.ApproximationRank);
+            Assert.Equal(rightMatrix.Count, model.Model.NumberOfColumns * model.Model.ApproximationRank);
+            Assert.Equal(leftMatrix[0], (double)0.3091519,5);
```

Comment:
>leftMatrix[0] [](start = 25, length = 13)

@wschin do you know why I get
on Mac:
 Expected: 0.41953 (rounded from 0.419525027275085)
Actual: 0.30915 (rounded from 0.3091519)
on Linux:
Expected: 0.51506 (rounded from 0.515059947967529)
Actual: 0.30915 (rounded from 0.3091519)

Reply:
and how you handle that discrepancy between mac/linux and windows before?

---
In reply to: [255225321](https://github.com/dotnet/machinelearning/pull/2451#discussion_r255225321) [](ancestors = 255225321)